Modules/Common/PolkadotLikeMinimalModule.php

```diff
@@ -0,0 +1,240 @@
+{
+    use UTXOTraits;
+
+    public ?BlockHashFormat $block_hash_format = BlockHashFormat::HexWith0x;
+    public ?AddressFormat $address_format = AddressFormat::AlphaNumeric;
+    public ?TransactionHashFormat $transaction_hash_format = TransactionHashFormat::HexWith0x; // We process transactions with
+    // transfers only, so we don't take extrinsics without hashes into account
+    public ?TransactionRenderModel $transaction_render_model = TransactionRenderModel::Even;
+    public ?CurrencyFormat $currency_format = CurrencyFormat::Static;
+    public ?CurrencyType $currency_type = CurrencyType::FT;
+    public ?FeeRenderModel $fee_render_model = FeeRenderModel::ExtraF;
+    public ?bool $hidden_values_only = false;
+
+    public ?array $events_table_fields = ['block', 'transaction', 'sort_key', 'time', 'address', 'effect', 'failed', 'extra'];
+    public ?array $events_table_nullable_fields = ['extra'];
+
+    public ?ExtraDataModel $extra_data_model = ExtraDataModel::Default;
+
+    public ?bool $should_return_events = true;
+    public ?bool $should_return_currencies = false;
+    public ?bool $allow_empty_return_events = true;
+
+    public ?bool $mempool_implemented = false;
+    public ?bool $forking_implemented = true;
+
+    //
+
+    public ?array $block_data = null;
+
+    //
+
+    final public function pre_initialize()
+    {
+        $this->version = 1;
+    }
+
+    final public function post_post_initialize()
+    {
+        //
+    }
+
+    final public function inquire_latest_block()
+    {
+        return (int)requester_single($this->select_node(),
+            endpoint: 'api/scan/metadata',
+            params: true,
+            result_in: 'data',
+            timeout: $this->timeout)['blockNum'];
+    }
+
+    final public function ensure_block($block_id, $break_on_first = false)
+    {
+        $this->block_data = requester_single($this->select_node(),
+            endpoint: 'api/scan/block',
+            params: ['block_num' => $block_id],
+            result_in: 'data',
+            timeout: $this->timeout);
+
+        $this->block_hash = $this->block_data['hash'];
+        $this->block_time = date('Y-m-d H:i:s', (int)$this->block_data['block_timestamp']);
+    }
+
+    final public function pre_process_block($block_id)
+    {
+        if ($block_id === 0)
+        {
+            $this->set_return_events([]);
+            return;
+        }
+
+        $extrinsics = $extrinsics_with_hashes = [];
+
+        foreach ($this->block_data['extrinsics'] as $extrinsic)
+            if ($extrinsic['extrinsic_hash'])
+                $extrinsics_with_hashes[] = $extrinsic['extrinsic_hash'];
+
+        if (!$extrinsics_with_hashes)
+        {
+            $this->set_return_events([]);
+            return;
+        }
+
+        $events = [];
+        $multi_curl = [];
+
+        foreach ($extrinsics_with_hashes as $extrinsic_hash)
+            $multi_curl[] = requester_multi_prepare($this->select_node(),
```

Comment:
oh, this is the wrong way, as the `hash` depends on extrisics body, i.e. there can be many extrinsics with the same hash.
Why don't you use block_id-extrinsic_id,i.e. extrinsic_index.
Also subscan API returns a single extrinsic for the specified hash, returning the array of extrinsics for the `hash` was one of our customization.

Anyway using current strategy with `hash` leads to reprocess extrinsics, that were already processed.
Use `extrinsic_index` instead of `hash`, as [it's a unique key](https://wiki.polkadot.network/docs/build-protocol-info#unique-identifiers-for-extrinsics)